trunk/SUAVE/Components/Energy/Converters/Propeller.py

```diff
@@ -86,611 +83,330 @@ def spin(self,conditions):
         epsilon[epsilon==np.inf] = 10. 
-        deltar   = (r[1]-r[0])
-        thrust   = rho*B*(np.sum(Gamma*(Wt-epsilon*Wa)*deltar,axis=1)[:,None])
-        torque   = rho*B*np.sum(Gamma*(Wa+epsilon*Wt)*r*deltar,axis=1)[:,None]
+        deltar                   = (r[1]-r[0])  
+        
+        blade_T_distribution     = rho*(Gamma*(Wt-epsilon*Wa))*deltar 
+        blade_Q_distribution     = rho*(Gamma*(Wa+epsilon*Wt)*r)*deltar 
+        thrust                   = rho*B*(np.sum(Gamma*(Wt-epsilon*Wa)*deltar,axis=1)[:,None])
+        torque                   = rho*B*np.sum(Gamma*(Wa+epsilon*Wt)*r*deltar,axis=1)[:,None] 
+        power                    = omega*torque 
+        Va_2d                    = np.repeat(Wa.T[ : , np.newaxis , :], Na, axis=1).T
+        Vt_2d                    = np.repeat(Wt.T[ : , np.newaxis , :], Na, axis=1).T
+        Va_ind_2d                = np.repeat(va.T[ : , np.newaxis , :], Na, axis=1).T
+        Vt_ind_2d                = np.repeat(vt.T[ : , np.newaxis , :], Na, axis=1).T
+        blade_T_distribution_2d  = np.repeat(blade_T_distribution.T[ np.newaxis,:  , :], Na, axis=0).T 
+        blade_Q_distribution_2d  = np.repeat(blade_Q_distribution.T[ np.newaxis,:  , :], Na, axis=0).T 
+        
+        blade_Gamma_2d           = np.repeat(Gamma.T[ : , np.newaxis , :], Na, axis=1).T
+        blade_dT_dR              = rho*(Gamma*(Wt-epsilon*Wa))
+        blade_dT_dr              = rho*(Gamma*(Wt-epsilon*Wa))*R
+        blade_dQ_dR              = rho*(Gamma*(Wa+epsilon*Wt)*r)
+        blade_dQ_dr              = rho*(Gamma*(Wa+epsilon*Wt)*r)*R
+        
+        Vt_ind_avg = vt
+        Va_ind_avg = va
+        Vt_avg     = Wt
+        Va_avg     = Wa
+          
+        # calculate coefficients 
         D        = 2*R 
+        Cq       = torque/(rho*(n*n)*(D*D*D*D*D)) 
         Ct       = thrust/(rho*(n*n)*(D*D*D*D))
-        Ct[Ct<0] = 0.        #prevent things from breaking
-        kappa    = self.induced_power_factor 
-        Cd0      = self.profile_drag_coefficient   
-        Cp       = np.zeros_like(Ct)
-        power    = np.zeros_like(Ct)        
-        for i in range(len(Vv)):
-            if -1. <Vv[i][0] <1.: # vertical/axial flight
-                Cp[i]       = (kappa*(Ct[i]**1.5)/(2**.5))+sigma*Cd0/8.
-                power[i]    = Cp[i]*(rho[i]*(n[i]*n[i]*n[i])*(D*D*D*D*D))
-                torque[i]   = power[i]/omega[i]  
-            else:  
-                power[i]    = torque[i]*omega[i]   
-                Cp[i]       = power[i]/(rho[i]*(n[i]*n[i]*n[i])*(D*D*D*D*D))
-
-        # torque coefficient 
-        Cq = torque/(rho*(n*n)*(D*D*D*D)*R) 
-        
-        thrust[conditions.propulsion.throttle[:,0] <=0.0] = 0.0
-        power[conditions.propulsion.throttle[:,0]  <=0.0] = 0.0 
-        torque[conditions.propulsion.throttle[:,0]  <=0.0] = 0.0 
-        thrust[omega<0.0] = - thrust[omega<0.0] 
+        Cp       = power/(rho*(n*n*n)*(D*D*D*D*D))  # correct 
```

Comment:
does this comment mean that it is done correctly, or it needs correcting?

Reply:
This comment should be removed.  It is done correctly. Issue resolved